tests/test_spotirec.py

```diff
@@ -0,0 +1,1938 @@
+        spotirec.parse_seed_info(['vapor-death-jazz'])
+        sys.stdout.close()
+        sys.stdout = self.stdout_preserve
+        with open(self.test_log, 'r') as f:
+            stdout = f.read()
+            self.assertIn(expected, stdout)
+
+    @ordered
+    def test_add_to_blacklist(self):
+        """
+        Testing add_to_blacklist()
+        """
+        spotirec.add_to_blacklist(['spotify:track:testtrack'])
+        blacklist = spotirec.conf.get_blacklist()
+        self.assertIn('spotify:track:testtrack', blacklist['tracks'].keys())
+        self.assertDictEqual(blacklist['tracks']['spotify:track:testtrack'],
+                             {'name': 'track0', 'uri': 'spotify:track:testtrack', 'artists': ['frankie0', 'frankie1']})
+        spotirec.conf.remove_from_blacklist('spotify:track:testtrack')
+        blacklist = spotirec.conf.get_blacklist()
+        self.assertDictEqual(blacklist['tracks'], {})
+
+    @ordered
+    def test_remove_from_blacklist(self):
+        """
+        Testing remove_from_blacklist()
+        """
+        spotirec.conf.add_to_blacklist(self.test_track0, 'spotify:track:testid0')
+        spotirec.conf.add_to_blacklist(self.test_artist0, 'spotify:artist:testid0')
+        blacklist = spotirec.conf.get_blacklist()
+        self.assertIn('spotify:track:testid0', blacklist['tracks'].keys())
+        self.assertIn('spotify:artist:testid0', blacklist['artists'].keys())
+        spotirec.remove_from_blacklist(['spotify:track:testid0', 'spotify:artist:testid0'])
+        blacklist = spotirec.conf.get_blacklist()
+        self.assertNotIn('spotify:track:testid0', blacklist['tracks'].keys())
+        self.assertNotIn('spotify:artist:testid0', blacklist['artists'].keys())
+
+    @ordered
+    def test_print_blacklist(self):
+        """
+        Testing print_blacklist()
+        """
+        spotirec.conf.add_to_blacklist(self.test_track0, 'spotify:track:testid0')
+        spotirec.conf.add_to_blacklist(self.test_artist0, 'spotify:artist:testid0')
+        spotirec.print_blacklist()
+        sys.stdout.close()
+        sys.stdout = self.stdout_preserve
+        with open(self.test_log, 'r') as f:
+            stdout = f.read()
+            self.assertIn('Tracks', stdout)
+            self.assertIn('Artists', stdout)
+            self.assertIn('test0 by frankie0 - spotify:track:testid0', stdout)
+            self.assertIn('frankie0 - spotify:artist:testid0', stdout)
+
+    @ordered
+    def test_generate_img(self):
+        """
+        Testing generate_img()
+        """
+        img = spotirec.generate_img(['test:test:test', 'test:test:test', 'test:test:test'])
+        # hash: 2eccd587915e21ab37d6352bb55cfc8754545daa6ba1c3be0b759d66fbb36acb
+        # color: [46, 204, 213]
+        self.assertTrue(any(x[1] == (46, 204, 213) for x in img.getcolors()))
+        self.assertTrue(any(x[1] == (200, 200, 200) for x in img.getcolors()))
+        self.assertEqual(img.size, (320, 320))
```

Comment:
I get that just checking that the colours exist in the image is quicker - but why not hashing? Just for good measure :man_shrugging:

Reply:
Fixed in 47736fd